src/EnergyPlus/HeatBalanceSurfaceManager.cc

```diff
@@ -8874,4 +8939,100 @@ void GatherComponentLoadsSurfAbsFact(EnergyPlusData &state)
     }
 }
 
+void GetGroundSurfacesTemperatureAverage(EnergyPlusData &state)
+{
+    //  returns ground surfaces average temperature (deg C)
+    //  ground surfaces viewed by a building exterior surface
+    //  ground surfaces temperature weighed using view factors
+
+    // local vars
+    Real64 GndSurfaceTemp;
+    Real64 GndSurfViewFactor;
+    Real64 GndSurfaceTempSum;
+
+    if (!state.dataGlobal->AnyLocalEnvironmentsInModel) {
+        return;
+    }
+
+    for (int SurfNum = 1; SurfNum <= state.dataSurface->TotSurfaces; ++SurfNum) {
+        if (!state.dataSurface->IsSurfPropertyGndSurfacesDefined(SurfNum)) continue;
+        auto &GndSurfsProperty = state.dataSurface->GroundSurfsProperty(state.dataSurface->GroundSurfsPropertyNum(SurfNum));
+        if (GndSurfsProperty.SurfsViewFactorSum == 0.0) {
+            state.dataSurface->UseSurfPropertyGndSurfTemp(SurfNum) = false;
+            continue;
+        }
+        GndSurfaceTemp = 0.0;
+        GndSurfViewFactor = 0.0;
+        GndSurfaceTempSum = 0.0;
+        for (int gSurfNum = 1; gSurfNum <= GndSurfsProperty.NumGndSurfs; gSurfNum++) {
+            GndSurfViewFactor = GndSurfsProperty.GndSurfs(gSurfNum).ViewFactor;
+            if (GndSurfViewFactor == 0.0) continue;
+            if (GndSurfsProperty.GndSurfs(gSurfNum).TempSchPtr == 0) continue;
+            GndSurfaceTemp = ScheduleManager::GetCurrentScheduleValue(state, GndSurfsProperty.GndSurfs(gSurfNum).TempSchPtr);
+            GndSurfaceTempSum += GndSurfViewFactor * pow_4(GndSurfaceTemp + DataGlobalConstants::KelvinConv);
+        }
+        if (GndSurfaceTempSum == 0.0) {
+            GndSurfsProperty.SurfsTempAvg = 0.0;
+            state.dataSurface->UseSurfPropertyGndSurfTemp(SurfNum) = false;
+            continue;
+        }
+        GndSurfsProperty.SurfsTempAvg = root_4(GndSurfaceTempSum / GndSurfsProperty.SurfsViewFactorSum) - DataGlobalConstants::KelvinConv;
+    }
+}
```

Comment:
Function that calculates view factors weighted ground surfaces average temperature.

Reply:
I saw above that we are adding the Kelvin conversion back on when doing radiation calculations, as expected.  Is it reasonable to just keep the `SurfsTempAvg` variable in Kelvin and not have to subtract and add the conversion back on?  Is it used for a user facing aspect where Celsius is preferred?